xonsh/built_ins.py

```diff
@@ -613,6 +613,10 @@ def run_subproc(cmds, captured=True):
             'obj': prev_proc,
             'bg': background
         })
+    try:
+        builtins.__xonsh_shell__.settitle()
+    except AttributeError:
+        pass
```

Comment:
This block could be:

```python
if hasattr(builtins.__xonsh_shell__, 'settitle'):
    builtins.__xonsh_shell__.settitle()
```

rather than relying on exceptions.  Or, just remove any conditional and do a straight up `builtins.__xonsh_shell__.settitle()` and rely on duck typing.